File: src/main/java/com/musiqueando/domain/Cancion.java
package com.musiqueando.domain;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "canciones")
public class Cancion {
    @Id
    private String id;
    private List<String> acordes;
    private List<String> letra;
    private Integer cejilla;
    private String genero;
    private String nombre;
    private String artistaId;
}

File: app/src/main/java/com/example/hospitalcompose/domain/modelo/Cancion.kt
package com.example.hospitalcompose.domain.modelo

import kotlinx.serialization.Serializable
import kotlinx.serialization.SerialName

@Serializable
data class Cancion(
    @SerialName("_id")
    val id: String,
    val nombre: String,
    val artista: String,
    val genero: String,
    val cejilla: Int,
    val acordes: List<String>,
    val letra: List<String>
)
